util/util.c

```diff
@@ -280,3 +280,26 @@ uint32_t srt_to_uint32(const char *str)
 	else
 		return value;
 }
+
+void put_unaligned_be16(uint8_t *buf, uint16_t val)
+{
+	buf[1] = val & 0xFF;
```

Comment:
the implementation for all of this functions doesn't do what it says - you should find an implementation that converts to be16 depending on your processor endianess

the way it's implemented may work for you because you've tested it on a little endian processor so you perform the swap anyway, but there is a scenario where you don't need to perform the swap

Reply:
I would say that `val & 0xFF`  will give the same result for both le and be.
Same for `val >> 8`
Am I wrong?

Reply:
Similar functions are present on linux, so I would say that they are OK.

Reply:
ok i misunderstood what the functions should do, i had a look at linux counterparts and it seems this would work